### PR TITLE
Added cheatsheet for csharp

### PR DESCRIPTION
**Pull request by gitcodes**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [x] Cheat Sheets: **`New C# Cheat Sheet`**
    
###### Description of changes

Provide an overview of the changes this pull request introduces.

---

Instant Answer Page: https://duck.co/ia/view/csharp_cheat_sheet

---
https://duck.co/ia/view/csharp_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@gitcodes Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/csharp_cheat_sheet_mobile.png?yP8910HOus)

File: share/goodie/cheat_sheets/json/csharp.json [hunk 1]
```diff
@@ -0,0 +1,266 @@
+{
+	"id": "csharp_cheat_sheet",
+	"name": "Csharp",
```
Comment (gaulrobe) [inline share/goodie/cheat_sheets/json/csharp.json]: `C#` may be a more instantly identifiable name here.

File: share/goodie/cheat_sheets/json/csharp.json [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	"description": "C# is an object-oriented programming language from Microsoft that aims to combine the computing power of C++ with the programming ease of Visual Basic. C# is based on C++ and contains features similar to those of Java.",
```
Comment (gaulrobe) [inline share/goodie/cheat_sheets/json/csharp.json]: A shorter description that explains this cheat sheet here might be better. Something like `Common C# syntax`?

**Comment by gaulrobe** [comment]
Could you run the JSON here through something like https://jsonformatter.curiousconcept.com/ so that we get the formatting to be consistent?

**Comment by gaulrobe** [comment]
@gitcodes this IA will be very useful! Just a couple of comments, otherwise this starting to look good!

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/csharp_cheat_sheet_mobile.png?0NFvtcIg9S)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/csharp_cheat_sheet_mobile.png?w75HrePCRw)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/csharp_cheat_sheet_mobile.png?sVBSkdcTKe)

File: share/goodie/cheat_sheets/json/csharp.json [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+            "key":"f()",
+            "val":" function invocation."
+         },
+         {  
+            "key":"?[]",
+            "val":"null conditional indexing.Returns null if the left hand operand is null."
+         },
+         {  
+            "key":"++",
+            "val":"postfix increment."
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/csharp.json]: I'd say increment operator; as it depends if it's `++x` or `x++`

File: share/goodie/cheat_sheets/json/csharp.json [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+         },
+         {  
+            "key":"--",
+            "val":"postfix decrement."
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/csharp.json]: Again I'd say `decrement`

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/csharp_cheat_sheet_mobile.png?TzuAcSc0PR)

File: share/goodie/cheat_sheets/json/csharp.json [hunk 5]
```diff
@@ -0,0 +1,337 @@
+{  
+   "id":"csharp_cheat_sheet",
+   "name":"Csharp",
+   "description":"Common C# syntax ",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/csharp.json]: Lets lose the space at the end.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/csharp_cheat_sheet_mobile.png?5THJ7GR7fB)

**Comment by daxtheduck** [comment]
@gitcodes this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**